widget/view.controller.js

```diff
@@ -4,76 +4,104 @@
-      .controller('heatmap100Ctrl', heatmap100Ctrl);
+  angular
+    .module('cybersponse')
+    .controller('heatmap100Ctrl', heatmap100Ctrl);
 
-    heatmap100Ctrl.$inject = ['$scope', 'widgetUtilityService', 'heatmapService', 'config', 'modelMetadatasService', '$state'];
+  heatmap100Ctrl.$inject = ['$scope', 'widgetUtilityService', 'heatmapService', 'config', 'modelMetadatasService', '$state' , '$q'];
 
-    function heatmap100Ctrl($scope, widgetUtilityService, heatmapService, config, modelMetadatasService, $state) {
-      $scope.config = config;
-      $scope.processing = true;
-      $scope.pageState = $state;
+  function heatmap100Ctrl($scope, widgetUtilityService, heatmapService, config, modelMetadatasService, $state, $q) {
+    $scope.config = config;
+    $scope.processing = true;
+    $scope.pageState = $state;
+    $scope.multipleFieldsItemsData = [];
+    $scope.connectorOperationResponse = {};
 
-      function _handleTranslations() {
-        widgetUtilityService.checkTranslationMode($scope.$parent.model.type).then(function () {
-          $scope.viewWidgetVars = {
-            // Create your translating static string variables here
-          };
-        });
-      }
-
-      function init() {
-        // To handle backward compatibility for widget
-        _handleTranslations();
-        checkCurrentPage($scope.pageState);
-        $scope.picklistField =  $scope.config.picklistField;
-        $scope.multipleFieldsItems= $scope.config.multipleFieldsItems || [];
-        var moduleMetaData = modelMetadatasService.getMetadataByModuleType($scope.config.resource);
-        //to check if dataSource is present and fetch data from connector action or else from API query
-        if(moduleMetaData.dataSource){ 
-          executeConnectorAction(moduleMetaData.dataSource);
-        }
-        else{
-          executeAPIQuery();
-        }  
-      }
+    function _handleTranslations() {
+      widgetUtilityService.checkTranslationMode($scope.$parent.model.type).then(function () {
+        $scope.viewWidgetVars = {
+          // Create your translating static string variables here
+        };
+      });
+    }
 
-      function checkCurrentPage(state){
-        if (state.current.name.includes('viewPanel.modulesDetail')) {
-          let params = $scope.pageState.current.params;
-          $scope.indicator = params.id;
+    function init() {
+      // To handle backward compatibility for widget
+      _handleTranslations();
+      checkCurrentPage($scope.pageState);
+      $scope.picklistField =  $scope.config.picklistField;
+      $scope.multipleFieldsItems= $scope.config.multipleFieldsItems || [];
+      var moduleMetaData = modelMetadatasService.getMetadataByModuleType($scope.config.resource);
+      //to check if dataSource is present and fetch data from connector action or else from API query
+      if(moduleMetaData.dataSource){ 
+        var _promises = [executeConnectorAction(moduleMetaData.dataSource)];
```

Comment:
what is a diff between executeConnectorAction and executeConnectorOperation?

Reply:
executeConnectorAction - to call connector action from data source 
executeConnectorOperation - to call connector operation from data source